### PR TITLE
scheme48: Add test.

### PR DESCRIPTION
**Pull request by woodruffw**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

Tests the output of a trivial Scheme program.

### CODE REVIEW

File: Formula/scheme48.rb [hunk 1]
```diff
@@ -21,4 +21,12 @@ def install
     system "make"
     system "make", "install"
   end
+
+  test do
+    (testpath/"hello.scm").write <<-EOS.undent
+      (display "Hello, World!") (newline)
+    EOS
+
+    assert_equal "Hello, World!\#{Unspecific}\n\n\#{Unspecific}\n\n", shell_output("#{bin}/scheme48 -a batch < hello.scm")
```
Comment (mikemcquaid) [inline Formula/scheme48.rb]: Would probably use `EOS.undent` rather than embedding newlines here.

File: Formula/scheme48.rb [hunk 2]
```diff
@@ -21,4 +21,19 @@ def install
...
+
+  test do
+    (testpath/"hello.scm").write <<-EOS.undent
+      (display "Hello, World!") (newline)
+    EOS
+
+    expected = <<-EOS.undent
+      Hello, World!\#{Unspecific}
+
+      \#{Unspecific}
```
Comment (UniqMartin) [inline Formula/scheme48.rb]: What's the meaning behind this and the above `\#{Unspecific}` output? And what's going on with the many newlines?

Comment (woodruffw) [inline Formula/scheme48.rb]: This particular interpreter doesn't have a file input argument, so it has to be fed lines of Scheme one-by-one in batch mode. AFAICT, the superfluous newlines and `#{Unspecifc}`s are the product of that one-by-one interpretation.

The other option is to generate an image file and load the interpreter with that, but that's a 3-4 step process compares to the two steps here (write file, execute).

Comment (UniqMartin) [inline Formula/scheme48.rb]: Makes sense. Thank you for the explanation!

**Comment by UniqMartin** [comment]
Merged. Thank you for yet another contribution to Homebrew, @woodruffw! :tada:

File: Formula/scheme48.rb (hunk 2 above)
Comment (woodruffw) [inline Formula/scheme48.rb]: No problem!

**Pull request merged**